### PR TITLE
service_account: Use go code to generate service account key pairs

### PR DESCRIPTION
**Pull request by yifan-gu**
Previously, the service account key pairs are generated by terraform in the bootkube module.

### CODE REVIEW

**Comment by wking** [comment]
Is the move from Terraform to Go just to make it easier for callers to provide their own keys?

**Comment by yifan-gu** [comment]
@wking Yeah, it will be easier, so the goal is not to use terraform for any assets generation.

**Comment by yifan-gu** [comment]
Looks like the cluster can't bootup now, debugging.

**Comment by abhinavdahiya** [comment]
```console
core@adahiya-0-master-0 /opt/tectonic/tls $ cat service-account.pub
-----BEGIN PUBLIC KEY-----
MIIBCgKCAQEA4LlFhUglModjo2ck78s1qEimYLgQOiE88mzUOfw7ajEcTR/zhzhD
Dd6EGOUNHDHyPyedhaIrL4zUkB25hk61mbxEnnzi/44UiLyFJHa4gqWc//BPyhrD
J54CXZWPdpdgOLfdjDb/3Vdv5V2/y6tO9AtPgqVB8DxAArv8vHb7bZrpjx8cOHfV
WNE5Y8Wa+IVVAtfwfgGkApdKXPEp9FUpPC/tERddwZ/fXvI0+Mogy+z40saHAhKZ
HAMWOU/ePe16b0xUKeXHlG9JZ3or++wFPlmATVa3WaYdHL9K6aJL0MWZ32u33lHE
+eRR77TxO+nVI2yHAcXTFWcVgvE2WoRW4QIDAQAB
-----END PUBLIC KEY-----
core@adahiya-0-master-0 /opt/tectonic/tls $ openssl rsa -in service-account.key -pubout
writing RSA key
-----BEGIN PUBLIC KEY-----
MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA4LlFhUglModjo2ck78s1
qEimYLgQOiE88mzUOfw7ajEcTR/zhzhDDd6EGOUNHDHyPyedhaIrL4zUkB25hk61
mbxEnnzi/44UiLyFJHa4gqWc//BPyhrDJ54CXZWPdpdgOLfdjDb/3Vdv5V2/y6tO
9AtPgqVB8DxAArv8vHb7bZrpjx8cOHfVWNE5Y8Wa+IVVAtfwfgGkApdKXPEp9FUp
PC/tERddwZ/fXvI0+Mogy+z40saHAhKZHAMWOU/ePe16b0xUKeXHlG9JZ3or++wF
PlmATVa3WaYdHL9K6aJL0MWZ32u33lHE+eRR77TxO+nVI2yHAcXTFWcVgvE2WoRW
4QIDAQAB
-----END PUBLIC KEY-----
core@adahiya-0-master-0 /opt/tectonic/tls $
```
The generated public key doesn't match the actual public key as shown by openssl.

**Comment by yifan-gu** [comment]
@abhinavdahiya Looks like I need to use ` x509.MarshalPKIXPublicKey()`

File: tests/run.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ SMOKE_TEST_OUTPUT="Never executed. Problem with one of previous stages"
 [ -z ${LICENSE_PATH+x} ] && (echo "Please set LICENSE_PATH"; exit 1)
 [ -z ${PULL_SECRET_PATH+x} ] && (echo "Please set PULL_SECRET_PATH"; exit 1)
 [ -z ${DOMAIN+x} ] && DOMAIN="tectonic-ci.de"
-[ -z ${AWS_REGION+x} ] && AWS_REGION="eu-west-1"
+[ -z ${AWS_REGION+x} ] && AWS_REGION="us-west-1"
```
Comment (wking) [inline tests/run.sh]: There are a few other occurrences of `eu-west-1`:

```console
$ git grep eu-west origin/pr/89
origin/pr/89:examples/tectonic.aws.yaml:    # `{ eu-west-1a = "10.0.0.0/20", eu-west-1b = "10.0.16.0/20" }`
origin/pr/89:examples/tectonic.aws.yaml:  region: eu-west-1
origin/pr/89:examples/tectonic.aws.yaml:    # Example: `{ eu-west-1a = "10.0.64.0/20", eu-west-1b = "10.0.80.0/20" }`
origin/pr/89:installer/pkg/config-generator/fixtures/test-aws.yaml:  region: eu-west-1
origin/pr/89:installer/pkg/config/aws/aws.go:   DefaultRegion = "eu-west-1"
origin/pr/89:installer/pkg/workflow/fixtures/terraform.tfvars:  "tectonic_aws_region": "eu-west-1",
origin/pr/89:steps/variables-aws.tf:`{ eu-west-1a = "10.0.0.0/20", eu-west-1b = "10.0.16.0/20" }`
origin/pr/89:steps/variables-aws.tf:Example: `{ eu-west-1a = "10.0.64.0/20", eu-west-1b = "10.0.80.0/20" }`
origin/pr/89:tests/jenkins-jobs/maintenance/tag_clean_aws_grafiti_job.groovy:    "ap-northeast-1" "eu-central-1" "eu-west-1" "eu-west-2" "sa-east-1" )
```

You may want to update at least the the `installer/` instances as well to stay consistent.

**Pull request merged**